archivessnake/create_daos.py

```diff
@@ -0,0 +1,205 @@
+#!/usr/bin/env python3
+
+"""
+Using DIP information from Archivematica, adds digital object to file-level children of a series in ArchivesSpace.
+
+Expects an external pickle file that contains a list of dictionaries with the following information:
+    current_full_path: full path to DIP (from Archivematica)
+    related_packages: AIP URI (from Archivematica)
+    uuid: DIP UUID (from Archivematica)
+    box: parsed from current_full_path
+    folder: parsed from current_full_path
+"""
+
+import argparse
+import pickle
+from configparser import ConfigParser
+from os.path import join
+
+import requests
+from asnake.aspace import ASpace
+
+# set up archivesspace client
+config = ConfigParser()
+config.read("local_settings.cfg")
+
+parser = argparse.ArgumentParser(
+    description="Using DIP information from Archivematica, adds DAOs to file-level children of a series in ArchivesSpace")
+parser.add_argument(
+    'resource', help='ArchivesSpace ID of resource. Example: 123')
+parser.add_argument('series', help='ArchivesSpace ID of series. Example: 4321')
+parser.add_argument(
+    'dip_info_file',
+    help='Path to pickle file containing array of dictionaries with DIP information.')
+args = parser.parse_args()
+
+
+def main(resource, series, dip_file):
+    """Main function, which is run when this script is executed"""
+    aspace = ASpace(
+        baseurl=config.get(
+            "ArchivesSpace", "baseURL"), username=config.get(
+            "ArchivesSpace", "username"), password=config.get(
+                "ArchivesSpace", "password"))
+    dao_url = join(config.get("ArchivesSpace", "baseURL"),
+                   "repositories/2/digital_objects")
+    dip_data = load_dip_file(dip_file)
+    for uri in get_uris(aspace, resource, series):
```

Comment:
I would maybe call this `component_uri` to distinguish it from the other `uris` in this method.